AWSLambdaS3Trigger/Function.cs

```diff
@@ -0,0 +1,173 @@
+    /// This method is called for every Lambda invocation. This method takes in an S3 event object and can be used 
+    /// to respond to S3 notifications.
+    /// </summary>
+    /// <param name="evnt"></param>
+    /// <param name="context"></param>
+    /// <returns></returns>
+    public async Task FunctionHandler(S3Event evnt, ILambdaContext context)
+    {
+        var eventRecords = evnt.Records ?? new List<S3Event.S3EventNotificationRecord>();
+        foreach (var record in eventRecords)
+        {
+            var s3Event = record.S3;
+            if (s3Event == null)
+            {
+                continue;
+            }
+
+            try
+            {
+                //String RDSConnectionString = await GetSecret();
+
+                var response = await this.S3Client.GetObjectAsync(s3Event.Bucket.Name, s3Event.Object.Key);
+                using (var stream = new MemoryStream())
+                {
+
+                    await response.ResponseStream.CopyToAsync(stream);
+                    stream.Position = 0;
+                    var workbook = new XSSFWorkbook(stream); // For .xlsx files
+
+                    ISheet sheet = workbook.GetSheetAt(0);
+                    int rowCount = sheet.PhysicalNumberOfRows;
+
+                    for (int row = 1; row < rowCount; row++)
+                    {
+                        IRow dataRow = sheet.GetRow(row);
+
+                        var toDo = new Entities_ADO.Models.ToDo()
+                        {
+                            Title = dataRow.GetCell(1).ToString(),
+                            IsCompleted = Convert.ToBoolean(dataRow.GetCell(2).ToString()),
+                            Owner = Convert.ToInt32(dataRow.GetCell(3).ToString())
+                        };
+
+                        var sqlConnection = new SqlConnection("server=myrds.c43r0x0dmovz.us-east-1.rds.amazonaws.com,1433;Initial Catalog=Lab_RDS;Integrated Security=False;Encrypt=False;User Id=admin;Password=admin1234");
+                        sqlConnection.Open();
+
+                        using (var command = sqlConnection.CreateCommand())
+                        {
+                            command.CommandType = CommandType.StoredProcedure;
+                            command.CommandText = "INSERT_TODOS";
+
+                            SqlParameter Id = new SqlParameter();
+                            Id.ParameterName = "@Id";
+                            Id.SqlDbType = SqlDbType.Int;
+                            Id.Direction = ParameterDirection.Output;
+                            command.Parameters.Add(Id);
+
+                            SqlParameter Title = new SqlParameter();
+                            Title.ParameterName = "@Title";
+                            Title.SqlDbType = SqlDbType.NVarChar;
+                            Title.Direction = ParameterDirection.Input;
+                            Title.Value = toDo.Title;
+                            command.Parameters.Add(Title);
```

Comment:
we can declare SqlParameter variables such as:

```
var title = new new SqlParameter(){
ParameterName = "@Title",
SqlDbType = SqlDbType.NVarChar,
Direction = ParameterDirection.Input,
}
```

Reply:
Its just a matter of personal preferences. There is no difference in code performance or efficiency.